packages/flutter_tools/lib/src/template.dart

```diff
@@ -328,7 +328,24 @@ class Template {
           context['androidIdentifier'] = _escapeKotlinKeywords(androidIdentifier);
         }
 
-        final String renderedContents = _templateRenderer.renderString(templateContents, context);
+        Map<String, Object?> localContext = context;
+
+        // YAML files require that their input is properly escaped with quotes.
+        // Adjust the current context to escape the needed inputs, before writing to the template.
+        if (finalDestinationFile.path.endsWith('.yaml')) {
+          if (finalDestinationFile.path.endsWith('pubspec.yaml')) {
+            // Grab a copy of the context, since it is used to render other templates as well.
+            localContext = Map<String, Object?>.of(localContext);
+
+            final String? description = localContext['description'] as String?;
+
+            if (description != null && description.isNotEmpty) {
+              localContext['description'] = escapeYamlString(description);
+            }
+          }
+        }
```

Comment:
Is the second condition redundant with the first? Also, consider making the leading comment more concise.

```suggestion
        // Ensure inputs are escaped when necessary.
        if (finalDestinationFile.path.endsWith('.yaml')) {
          // Use copy of the context, since the original is used in rendering other templates.
          localContext = Map<String, Object?>.of(localContext);

          final String? description = localContext['description'] as String?;

          if (description != null && description.isNotEmpty) {
            localContext['description'] = escapeYamlString(description);
          }
        }
```

Reply:
No worries :)

Yeah, I was a bit conservative about "what if we need to handle other YAML files in the future", but that doesn't happen often.

Adjusted the condition and updated the comment.